### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by Sven-Lyco**
- #47

### CODE REVIEW

**Review by Ju-Schn — APPROVED** [review]
🚀  Veeery nice work!  🚀

File: api/getMovieDetails/index.mjs [hunk 1]
```diff
@@ -4,6 +4,7 @@ const { API_BASE_MOVIES_URL, API_KEY, API_LANGUAGE } = process.env;
 
 export default async function getMovieDetails(req, res) {
   if (req.method === 'POST') {
+    // To test POST request with _getMovieDetails.http use: const { id } = req.body;
```
Comment (macfrei) [inline api/getMovieDetails/index.mjs:7]: Is this the right place for this comment? Also, the comment does not really tell me what to do exactly. Maybe rephrase it or leave it out? Your code is quite understandable.

File: src/components/AddButton.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import AddButton from './AddButton';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+describe('AddButton', () => {
+  it('renders a button', () => {
+    const onClick = jest.fn();
+    render(<AddButton onClick={onClick} />);
+
+    const button = screen.getByRole('button');
```
Comment (macfrei) [inline src/components/AddButton.spec.js:10]: Maybe add the `{ name: '...' }` to be even more specific?

File: src/components/ButtonCheck.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  it('has onClick function', () => {
+    const onClick = jest.fn();
+    render(<ButtonCheck onClick={onClick} isActive={false} />);
+
+    const button = screen.getByRole('button');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+});
```
Comment (macfrei) [inline src/components/ButtonCheck.spec.js:22]: There are two tests missing: if isActive is true, then the name of the button is 'gesehen', if it is false, then it's 'ungesehen'.

File: src/components/CastList.js [hunk 4]
```diff
@@ -1,33 +1,31 @@
 import styled from 'styled-components';
-import PosterActor from './PosterActor';
 import defaultPoster from '../assets/images/profile.png';
+import PosterActor from './PosterActor';
 
 export default function CastList({ castList, listName }) {
   return (
     <>
-      <ListHeader>{listName}</ListHeader>
+      <ListHeader>{listName ? listName : ''}</ListHeader>
```
Comment (macfrei) [inline src/components/CastList.js:8]: For accessibility reasons I don't think it is a good idea to leave an empty string within your `h3`. Headings are landmarks and if there is an empty landmark, it doesn't really make sense. I think it would be better to either render the `h3` or not at all:
```suggestion
      {listName && <ListHeader>{listName}</ListHeader>}
```

File: src/components/CastList.js [hunk 5]
```diff
@@ -1,33 +1,31 @@
...
       {castList?.length >= 1 ? (
-        <StyledList role="list" listLength={castList?.length}>
+        <List role="list" listLength={castList?.length}>
```
Comment (macfrei) [inline src/components/CastList.js:10]:
```suggestion
        <List role="list" listLength={castList.length}>
```
You are only rendering this, if the `castList.length >= 1` and then it should have a length. I don't think you need the optional chaining here.

File: src/components/CastList.js [hunk 6]
```diff
@@ -1,33 +1,31 @@
...
           {castList?.map(({ id, name, character, profile_path }) => (
```
Comment (macfrei) [inline src/components/CastList.js:11]: Same here:
```suggestion
          {castList.map(({ id, name, character, profile_path }) => (
```

File: src/components/DeleteButton.spec.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import DeleteButton from './DeleteButton';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+describe('DeleteButton', () => {
+  it('renders a button', () => {
+    const onClick = jest.fn();
+    render(<DeleteButton onClick={onClick} />);
+
+    const button = screen.getByRole('button');
```
Comment (macfrei) [inline src/components/DeleteButton.spec.js:10]: Same with the `name`.

File: src/components/PosterList.js [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
 import defaultPoster from '../assets/images/poster.png';
+import Poster from './Poster';
 
 export default function PosterList({ list, listName }) {
   return (
     <>
-      <ListHeader>{listName}</ListHeader>
-      <StyledList role="list" listLength={list?.length}>
-        {list.map(({ name, title, poster_path, id }) => (
+      <ListHeader>{listName ? listName : ''}</ListHeader>
```
Comment (macfrei) [inline src/components/PosterList.js:9]: Same here 🙂

File: src/components/ProviderList.js [hunk 9]
```diff
@@ -17,7 +18,7 @@ export default function ProviderList({ providerList }) {
         </a>
       </ListHeader>
       {flatrateProviders ? (
-        <StyledList role="list">
+        <List role="list">
           {flatrateProviders?.map(
```
Comment (macfrei) [inline src/components/ProviderList.js:22]: Don't think you need the optional chaining here 😉

File: src/components/SearchResultCard.js [hunk 10]
```diff
@@ -25,20 +27,8 @@ export default function SearchResultCard({
...
-                    : 'kein Release Datum vorhanden'
-                }`
-              : `${
-                  firstAirDate
-                    ? firstAirDate.substr(0, 4)
-                    : 'kein Release Datum vorhanden'
-                }`}{' '}
-            - {mediaType === 'movie' ? 'Film' : 'Serie'}
-          </span>
+          <p>{date ? date?.substr(0, 4) : 'kein Release Datum vorhanden'}</p>
```
Comment (macfrei) [inline src/components/SearchResultCard.js:30]:
```suggestion
          <p>{date ? date.substr(0, 4) : 'kein Release Datum vorhanden'}</p>
```

**Review by macfrei — COMMENTED** [review]
Nice job! Just added a few things, mostly minor details. 👍

File: src/components/AddButton.spec.js (hunk 2 above)
Comment (Sven-Lyco) [reply src/components/AddButton.spec.js:10]: I tried to test it this way, but i have to write in my test:
`{name: "plus_icon.svg zur Watchlist hinzufügen"}` 
Would this be okay? Because there is the icon and a screenreaderonly text? 🤔

File: src/components/CastList.js (hunk 5 above)
Comment (Sven-Lyco) [reply src/components/CastList.js:10]: If i don't use optional chaining here, my App crashes. :-/

File: src/components/CastList.js (hunk 6 above)
Comment (Sven-Lyco) [reply src/components/CastList.js:11]: same here

File: src/components/ProviderList.js (hunk 9 above)
Comment (Sven-Lyco) [reply src/components/ProviderList.js:22]: and here too
